### DISCUSSION TITLE
Laravel cuts off generating @json in the blade mid-air

### DISCUSSION THREAD

**Discussion by verybigelephants** [laravel/framework#49134] [Q&A] [answered]
### Laravel Version

v10.30.1

### PHP Version

8.2.12

### Database Driver & Version

postgres 14.5 

### Description

Inn a nlade, when using `@json` i sometimes get correctly generated json, but when the json object is larger like this

```
		var wtf = @json([
			'active_task' => $active_task,
			'styles' => $styles,
			'tasks_max' => 10,
			'tasks_default' => 3
		]);
```

The generation stops in the mid of the json parsing, outputing just
```
<?php echo json_encode([

			'active_task' => $active_task, 'styles' => $styles, 'tasks_max' => 10) ?>;
```
and i get this error 

![image](https://github.com/laravel/framework/assets/3118985/4f349303-5835-4305-8f3d-9fd04a2666f1)

While doing it via json_encode directly works correctly

```
		var wtf = {{json_encode([
			'active_task' => $active_task,
			'styles' => $styles,
			'tasks_max' => 10,
			'tasks_default' => 3
		])}};
```

I have found these bug reports saying it should already be fixed with ymfony/http-foundation 6.1.1 but mine is 6.3.7 and i still get this error

https://github.com/laravel/framework/issues/42654
https://github.com/laravel/framework/issues/42681

### Steps To Reproduce

controller
```
	public function test(Request $request){
		return view('frontend.test', [
			'active_task' => null,
			'styles' => [
				[
					'id'=> 1,
					'name' => 'name1',
					'test1' => 1,
					'test2' => false
				],
				[
					'id'=> 2,
					'name' => 'name2',
					'test1' => 1,
					'test2' => false
				],
				[
					'id'=> 3,
					'name' => 'name3',
					'test2' => 2,
					'test3' => true
				],
			]
		]);
```
view
```
@extends('... your layout that has a section yield')

@section('scripts') 
	<script type="text/javascript">
		var wtf = {{json_encode([
			'active_task' => $active_task,
			'styles' => $styles,
			'tasks_max' => 10,
			'tasks_default' => 3
		])}};
	</script>
@endsection

```

**Comment by valorin** [comment, marked as answer]
It's not a good idea to use the blade directives like this - they are built using a basic Regex, and can't handle complex structures. Even the escaping brackets (`{{ ... }}`) is regex based and not recommended for multi-line or super complex expressions.

I'd recommend setting your array in a variable in your controller or a `<?php ... ?>` section at the start of your template and then rendering that variable using the `JS` facade: https://laravel.com/docs/10.x/blade#rendering-json

```
<?php
$array = [
    'active_task' => $active_task,
    'styles' => $styles,
    'tasks_max' => 10,
    'tasks_default' => 3
];
?>

<script>
var wtf = {{ Js::from($array) }};;
</script>
```
